addons/web/static/tests/_framework/mock_server/mock_model.js

```diff
@@ -120,7 +120,7 @@ const READ_GROUP_NUMBER_GRANULARITY = [
 
 const DATE_FORMAT = {
     day: (date) => date.toFormat("yyyy-MM-dd"),
-    day_of_week: (date) => date.weekday,
+    day_of_week: (date) => date.weekday % 7, // number of days after the first day of the week (assumed to be Sunday)
```

Comment:
Ideally, it should also be fixed in the legacy mock model (legacy/helpers/mock_server.js)